src/components/Calendar/src/Calendar.vue

```diff
@@ -121,214 +109,180 @@ export default {
 
 	props: {
 		/**
-		 * Language specific format to use
-		 */
-		locale: {
-			type: String,
-			default: 'en-US',
-		},
-		/**
-		 * Selected date value (YYYY-MM-DD)
+		 * Selected date value in ISO format.
 		 */
 		selectedDate: {
 			type: String,
-			default: '',
-			validator: isDateFormat,
+			default: undefined,
+			validator: isIsoFormat,
 		},
 		/**
-		 * Disable the dates before the min-date
+		 * Disable the dates before the min-date in ISO format.
 		 */
 		minDate: {
 			type: String,
-			default: '',
-			validator: isDateFormat,
+			default: undefined,
+			validator: isIsoFormat,
 		},
 		/**
-		 * Disable the dates after the max-date
+		 * Disable the dates after the max-date in ISO format.
 		 */
 		maxDate: {
 			type: String,
-			default: '',
-			validator: isDateFormat,
+			default: undefined,
+			validator: isIsoFormat,
 		},
 		/**
-		 * List of disabled dates
+		 * List of disabled dates in ISO format.
 		 */
 		disabledDates: {
 			type: Array,
 			default: () => ([]),
-			validator: (v) => v.every((date) => isDateFormat(date)),
+			validator: (disabledDates) => disabledDates.every((date) => isIsoFormat(date)),
+		},
+		/**
+		 * Calendar locale. Defaults to browser locale.
+		 */
+		locale: {
+			type: String,
+			default: undefined,
 		},
 	},
 
 	data() {
 		return {
-			viewingDate: this.selectedDateObj() || new Date(),
+			showingMonth: this.selectedDateObj || new Date(),
 		};
 	},
 
 	computed: {
 		monthName() {
-			return this.viewingDate.toLocaleString(this.locale, { month: 'long', year: 'numeric' });
+			return this.showingMonth.toLocaleString(this.locale, { month: 'long', year: 'numeric' });
 		},
 
 		weekdays() {
-			const date = new Date();
-			const firstDOW = startOfWeek(date);
-			const weekdays = [...new Array(7)].map((_, i) => addDays(firstDOW, i));
-			return weekdays.map((d) => d.toLocaleDateString(this.locale, { weekday: 'short' }));
+			const firstDayOfWeek = startOfWeek(new Date());
+			const weekdays = Array.from({ length: 7 }, (_, i) => addDays(firstDayOfWeek, i));
+			return weekdays.map((date) => date.toLocaleDateString(this.locale, { weekday: 'short' }));
 		},
 
 		weeks() {
-			const start = startOfMonth(this.viewingDate);
-			const end = lastDayOfMonth(this.viewingDate);
-			const eachDay = eachDayOfInterval({ start, end });
-			const emptyDays = [...new Array(getDay(eachDay[0]))];
-			const weeks = [];
+			const month = this.showingMonth.getMonth();
+			const dateOfMonth = startOfMonth(this.showingMonth);
 
-			[...emptyDays, ...eachDay].forEach((date, index) => {
-				const key = Math.floor(index / 7);
-				if (!weeks[key]) {
-					weeks[key] = [];
+			const weeks = [];
+			let week = [];
+			while (dateOfMonth.getMonth() === month) {
+				const weekday = dateOfMonth.getDay();
+				week[weekday] = new Date(dateOfMonth);
```

Comment:
I noticed a lot of the methods below were re-computing the Date object from a date string, so I decided to just use Date objects here.

This resulted in being able to remove methods `getDeriveViewingDateObj` and `deriveViewingDate`.